routers/repo/issue.go

```diff
@@ -1821,80 +1811,97 @@ func updatePullReviewRequest(ctx *context.Context) {
-					ctx.ServerError("issue.LoadRepo", err)
-					return
-				}
-
-				if err := issue.Repo.GetOwner(); err != nil {
-					ctx.ServerError("issue.Repo.GetOwner", err)
-					return
-				}
-
-				if issue.Repo.Owner.IsOrganization() {
-					team, err := models.GetTeamByID(-reviewID)
-					if err != nil {
-						ctx.ServerError("models.GetTeamByID", err)
-						return
-					}
-
-					if team.OrgID != issue.Repo.OwnerID {
-						ctx.Status(403)
-						return
-					}
-
-					err = isValidTeamReviewRequest(team, ctx.User, action == "attach", issue)
-					if err != nil {
-						ctx.Status(403)
-						return
-					}
+		if err := issue.LoadRepo(); err != nil {
+			ctx.ServerError("issue.LoadRepo", err)
+			return
+		}
 
-					err = issue_service.TeamReviewRequest(issue, ctx.User, team, action == "attach")
-					if err != nil {
-						ctx.ServerError("ReviewRequest", err)
-						return
-					}
+		if !issue.IsPull {
+			log.Warn(
+				"UpdatePullReviewRequest: refusing to add review request for non-PR issue %-v#%d",
+				issue.Repo, issue.Index,
+			)
+			ctx.Status(403)
+			return
+		}
+		if reviewID < 0 {
+			// negative reviewIDs represent team requests
+			if err := issue.Repo.GetOwner(); err != nil {
+				ctx.ServerError("issue.Repo.GetOwner", err)
+				return
+			}
 
-					continue
-				}
+			if !issue.Repo.Owner.IsOrganization() {
+				log.Warn(
+					"UpdatePullReviewRequest: refusing to add team review request for %s#%d owned by non organization UID[%d]",
+					issue.Repo.FullName(), issue.Index, issue.Repo.ID,
+				)
+				ctx.Status(403)
+				return
 			}
 
-			reviewer, err := models.GetUserByID(reviewID)
+			team, err := models.GetTeamByID(-reviewID)
 			if err != nil {
-				ctx.ServerError("GetUserByID", err)
+				ctx.ServerError("models.GetTeamByID", err)
 				return
 			}
 
-			err = isValidReviewRequest(reviewer, ctx.User, action == "attach", issue)
+			if team.OrgID != issue.Repo.OwnerID {
+				log.Warn(
+					"UpdatePullReviewRequest: refusing to add team review request for UID[%d] team %s to %s#%d owned by UID[%d]",
+					team.OrgID, team.Name, issue.Repo.FullName(), issue.Index, issue.Repo.ID)
+				ctx.Status(403)
+				return
+			}
+
+			err = isValidTeamReviewRequest(team, ctx.User, action == "attach", issue)
 			if err != nil {
+				log.Warn(
+					"UpdatePullReviewRequest: refusing to add invalid team review request for UID[%d] team %s to %s#%d owned by UID[%d]: Error: %v",
+					team.OrgID, team.Name, issue.Repo.FullName(), issue.Index, issue.Repo.ID,
+					err,
+				)
 				ctx.Status(403)
 				return
 			}
 
-			err = issue_service.ReviewRequest(issue, ctx.User, reviewer, action == "attach")
+			err = issue_service.TeamReviewRequest(issue, ctx.User, team, action == "attach")
 			if err != nil {
-				ctx.ServerError("ReviewRequest", err)
+				ctx.ServerError("TeamReviewRequest", err)
 				return
 			}
-		} else {
+			continue
+		}
+
+		reviewer, err := models.GetUserByID(reviewID)
+		if err != nil {
```

Comment:
```suggestion
		if err != nil {
			if models.IsErrUserNotExist(err) {
				log.Warn(
					"UpdatePullReviewRequest: Requested reviewer is not exist [reviewer_id : %d]",
					reviewID)
				ctx.Status(403)
				return
			}
```

How about this change? Thanks.

Reply:
Yeah seems reasonable - but try to follow the same style as the other warn messages I've added.